views/default/navigation/menu/user_hover/contents.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+// g = guid, pog = page_owner_guid, c = contexts, m = mac
+$guid = (int) get_input("g", 0, false);
+$page_owner_guid = (int) get_input("pog", 0, false);
+$contexts = (array) get_input("c", array(), false);
+$mac = get_input("m", "", false);
+$input = (array) get_input("i", array(), false);
+
+// verify MAC
+$data = serialize(array($guid, $page_owner_guid, $contexts, $input));
+if ($mac !== hash_hmac("sha256", $data, get_site_secret())) {
```

Comment:
Reminds me I need to make an HmacService https://github.com/Elgg/Elgg/issues/7824#issuecomment-71726594